packages/fetch-and-compile/lib/index.ts

```diff
@@ -0,0 +1,28 @@
+import debugModule from "debug";
+const debug = debugModule("fetch-and-compile");
+import type Config from "@truffle/config";
+import * as Recognizers from "./recognizers";
+import { fetchWithRecognizer } from "./fetch";
+import type * as Types from "./types";
+
+export { fetchWithRecognizer };
+
+export async function fetchSingle(
```

Comment:
I don't want this to be called just `fetchAndCompile` because like, this is just the single address case.  Do you think that case should be singled out like that?

Reply:
I think it's worth it. It's the name of the package, and it'll be the entrypoint that people commonly use. Makes sense to name it so it's never not obvious.

Reply:
OK, I wonder if this case should handle errors better then though. :P

Reply:
In what way are you thinking?

Reply:
Like, right now, if an error occurs during fetching, or during compilation, or if one of the fetchers doesn't support the network, it'll just throw; and the error it throws will be pretty nondescript.  So like possibly:

1. It should do something other than throwing?  I guess I'm not sure what that would be, so maybe not this.
2. Alternatively, maybe it should actually propagate the error instead of throwing its own less-informative one?  Although possibly it should only do that for compilation failure rather than fetch failure, since there's not exactly a hard distinction between fetch failure on the one hand and non-existent sources on the other.
3. Specifically in the case of a bad fetcher, maybe it should just swallow the error and continue trying with the others?

Reply:
I think we can start version 0.1.0 of this package without any robust error reporting